wp/lib/bap_wp/src/precondition.ml

```diff
@@ -626,11 +626,29 @@ let spec_inline (to_inline : Sub.t Seq.t) (sub : Sub.t) (_ : Arch. t)
   if Seq.mem to_inline sub ~equal:Sub.equal then
     Some {
       spec_name = "spec_inline";
-      spec = Inline
+      spec = Inline (Term.tid sub)
     }
   else
     None
 
+let spec_inline_plt (to_inline : Sub.t Seq.t) (sub : Sub.t) (_ : Arch.t)
```

Comment:
Comments here might be quite welcome, when we debug this in 3 months...

Reply:
This is not a PLT pattern, neither it is guaranteed that a function that has `@` symbol is a PLT (or any other) stub, nor that the `@` symbol will be used in the future. So this code is quite brittle and most likely will break in the future. 

The long story. In order to maintain the injectivity of the tied to name mapping, we have to ensure that all functions also have unique names. This invariant is maintained on the `Bap.Std.Program` level, and when a new function is added we check that it is unique and if it is not we mangle its name (using `@<addr>` is just one of the strategies that is used). See https://github.com/BinaryAnalysisPlatform/bap/issues/808 for more information on that, and here is the [code][1] that performs mangling. As you can see, the strategy can't guarantee that in case of two functions having the same name, we give the mangled name to the stub.

You might also interested in the [stub-resolver][2] ABI pass that @gitoleg is implementing as a part of  https://github.com/BinaryAnalysisPlatform/bap/pull/1036/ (Oleg, btw can you push this pass as a separate PR? As there are some questions about x86/primus part of that PR, but this particular module is independent of that and can be already merged.)



[1]: https://github.com/BinaryAnalysisPlatform/bap/blob/master/lib/bap_types/bap_ir.ml#L308-L341
[2]: https://github.com/BinaryAnalysisPlatform/bap/blob/84e74db3ae6fbf58c878d91eff7745a7700ef877/plugins/stub_resolver/stub_resolver_main.ml